Updated Security Considerations (#58)
suit-wg/suit-report: draft-ietf-suit-report.md

8 | Text Elements | {{capabilities}}
9 | Component Text Elements | {{capabilities}}

#  Security Considerations {#seccons

The SUIT\_Report serves four primary security objectives:

- Authenticity  

> - Integrity  

Author: While it is not important, I thought it would be good to talk about "who sent the SUIT report", "was it modified in transit", "has it been replayed" and "is the content available to others on route".

> - Integrity  

Reviewer: > I agree with this assessment @hannestschofenig. While I appreciate that a valid signature is both authenticity and integrity, these are separate properties that happen to both be provided by a signature.

Name one thing in object or transport security where integrity is lost but authenticity is retained.

- Replay protection  
- Confidentiality

The mechanisms for achieving these protections are outlined in {{container}}.

> Ideally, a SUIT\_Report SHOULD be conveyed as part of an attestation process,

Reviewer: ```suggestion
Ideally, a SUIT\_Report SHOULD be conveyed as part of a remote attestation process,
```

> such as embedded within an EAT token. This approach ensures that the SUIT\_Report

Reviewer: ```suggestion
such as embeding it in EAT tokens that represent RATS conceptual messages. This approach ensures that the SUIT\_Report
```

is cryptographically bound to the environment (hardware, software, or both) in
which it was generated, thereby strengthening its authenticity.

A SUIT\_Report may disclose sensitive information about the device on which they
were produced. In such cases, the SUIT\_Report MUST be encrypted, as specified in
{{container}}.

Furthermore, failure reports, particularly those involving cryptographic operations,
can unintentionally reveal insights into system weaknesses or vulnerabilities. As such,
SUIT\_Reports SHOULD be encrypted whenever possible, to minimize the risk of information
leakage.

In addition to these core security requirements, operational considerations must be taken
into account. When a SUIT\_Report is included within another protocol message (e.g., inside
an encrypted EAT token), care must be taken to avoid inadvertently leaking information and

> to uphold the principle of least privilege. For example, in many EAT-based attestation flows,

Reviewer: ```suggestion
to uphold the principle of least privilege. For example, in many EAT-based remote attestation flows,
```

the Verifier may not require the full SUIT\_Report. Similarly, the Relying Party might not
need access to it either.

To support least-privilege access, the SUIT\_Report should be independently encrypted, even
when the transport or enclosing token is also encrypted. This layered encryption ensures that
only authorized entities can access the contents of the SUIT\_Report.

> On the other hand, in scenarios where the EAT Verifier requires full access to the SUIT\_Report.

Reviewer: ```suggestion
In other scenarios, the EAT Verifier might require full access to a SUIT\_Report.
```

> For example, to extract or convert its content into specific EAT claims, such as `measres`

Reviewer: What is "its"? The message? The Verifier? I cannot parse this sentence.

Author: I believe you changed the sentence already to make it clear.
